product-scenarios/1-manage-public-partner-private-apis/1.1-expose-service-as-rest-api/1.1.2-create-rest-api-by-import-oas-document/src/test/java/org/wso2/am/scenario/tests/rest/api/creation/RESTApiCreationUsingOASDocTestCase.java

```diff
@@ -80,26 +114,95 @@ public void testCreateApiUsingValidOASDocumentFromJsonFile(String file_path) thr
+
+        apiRequest = new APIRequest(apiName, apiContext, apiVersion);
+        apiRequest.setSwagger(payloadJson);
+        HttpResponse designResponse2 = apiPublisher.designAPI(apiRequest);
+        verifyResponse(designResponse2);
 
-        String resource = (new JSONObject(serviceResponse.getData())).getJSONObject("api").get("resources").toString();
+        //Verify the API created with correct name, version and resources
+        String name = (new JSONObject(designResponse2.getData())).getJSONObject("data").get("apiName").toString();
+        String version = (new JSONObject(designResponse2.getData())).getJSONObject("data").get("version").toString();
+        Assert.assertEquals(name, apiName);
+        Assert.assertEquals(version, apiVersion);
+
+        HttpResponse getResponse = apiPublisher.getAPI(apiName, "admin", apiVersion);
+        String resource = (new JSONObject(getResponse.getData())).getJSONObject("api").get("resources").toString();
         Assert.assertTrue(resource != "null", "API resource was not imported correctly");
+    }
 
-        //clean the data
-        serviceResponse = apiPublisher.deleteAPI(apiName, apiVersion, "admin");
-        verifyResponse(serviceResponse);
+    @Test(description = "1.1.2.4", dependsOnMethods = "createApiWithValidOAS2DocumentAsYAMLFile")
+    public void createApiWithValidOAS3DocumentAsYAMLFile() throws Exception {
+        swagger_file = new File(resourceLocation + File.separator + "swaggerFiles/OAS3Document.yaml");
+
+        //Import api definition from swagger file
+        apiRequest = new APIRequest(import_definition_file, swagger_file.getAbsolutePath(), "", type);
+
+        HttpResponse designResponse1 = apiPublisher.designAPIWithOAS(apiRequest);
+        Assert.assertTrue(designResponse1.getData().contains("imported"));
+
+        //Save the API with imported values
+        String payload = readFromFile(swagger_file.getAbsolutePath());
+
+        Yaml yaml = new Yaml();
+        Map<String, Object> map = (Map<String, Object>) yaml.load(payload);
+        JSONObject json = new JSONObject(map);
+        String apiName = json.getJSONObject("info").get("title").toString();
+        String apiContext = json.get("basePath").toString();
+        String apiVersion = json.getJSONObject("info").get("version").toString();
+        String payloadJson = json.toString();
+
+        apiRequest = new APIRequest(apiName, apiContext, apiVersion);
+        apiRequest.setSwagger(payloadJson);
+        HttpResponse designResponse2 = apiPublisher.designAPI(apiRequest);
+        verifyResponse(designResponse2);
+
+        //Verify the API created with correct name, version and resources
+        String name = (new JSONObject(designResponse2.getData())).getJSONObject("data").get("apiName").toString();
+        String version = (new JSONObject(designResponse2.getData())).getJSONObject("data").get("version").toString();
+        Assert.assertEquals(name, apiName);
```

Comment:
For all asserts, provide a 3rd parameter "message", which will be printed in case if the assert fails.

Reply:
Fixed with 0b6782f997f735970123800b8f8409485f69c0de